src/model/position.js

```diff
@@ -782,6 +818,41 @@ export default class Position {
 }
 
 /**
+ * Method used to expose root setter to child classes.
+ * @protected
+ * @param {module:engine/model/position~Position} position Position of which root should be modified.
+ * @param {module:engine/model/element~Element|module:engine/model/documentfragment~DocumentFragment} root Root of the position.
+ */
+export function setRoot( position, root ) {
+	position[ _root ] = root;
+}
+
+/**
+ * Method used to expose path setter to child classes.
+ * @protected
+ * @param {module:engine/model/position~Position} position Position of which path should be modified.
+ * @param {Array.<Number>} path Position path. See {@link module:engine/model/position~Position#path}.
+ */
+export function setPath( position, path ) {
+	position[ _path ] = path;
+}
+
+// Helper for setting offset on give path array.
+// @private
+// @param {Array.<Number>} path Position path. See {@link module:engine/model/position~Position#path}.
+function getOffset( path ) {
+	return last( path ) || 0;
```

Comment:
Why `|| 0`? Wouldn't `undefined` be better?

Anyway, it is an incorrect situation if `path` has no items.

Reply:
I didn't want to change logic behind this AFAIR. It was:
```js
get offset() {
    return last( path ) || 0;
}
```
I don't know if it would have any side effects or not.

Reply:
Weird. Position always has to have an offset :S. Try if anything breaks without it ;).

Reply:
@scofalik: I was able to fail a test after removing `|| 0`:
```js
expect( new Position( root, [ false ] ) ).to.have.property( 'offset' ).that.equals( 0 );
expect( new Position( root, [ undefined ] ) ).to.have.property( 'offset' ).that.equals( 0 );
```

But this looks like a bad API use - so we might as well drop `|| 0` check as `position` accepts only non-empty array as path.

Reply:
Yeah, I'd remove `|| 0` and the test.